src/Authentication/Common.Authentication/Factories/AuthenticationFactory.cs

```diff
@@ -48,8 +48,18 @@ public AuthenticationFactory()
 
             TracingAdapter.Information(Resources.AdalAuthConfigurationTrace, configuration.AdDomain, configuration.AdEndpoint, 
                 configuration.ClientId, configuration.ClientRedirectUri, configuration.ResourceClientUri, configuration.ValidateAuthority);
+            IAccessToken token;
+            if (account.IsPropertySet(AzureAccount.Property.CertificateThumbprint))
```

Comment:
I think we should have a special check here for account.Type being ServicePrincipal as well.